journal.py

```diff
@@ -74,6 +76,16 @@ def modify(cls, eid=None, title=None, text=None):
         instance.text = text
         return instance
 
+    @property
+    def mkdown(self):
+        html_text = markdown(self.text, output_format='html5')
+
+        def my_highlight(matchobj):
+            return highlight(matchobj.group(0), PythonLexer(), HtmlFormatter())
+
+        pattern = r'(?<=<code>)[\s\S]*(?=<\/code>)'
+        return re.sub(pattern, my_highlight, html_text)
```

Comment:
oh, I so wish you'd just use the 'codehilite' and 'fenced_code' extensions to markdown.  But if this is working for you...

I just fear [parsing HTML with regexp](http://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags/1732454#1732454)